tests/org.jboss.tools.cdk.ui.bot.test/src/org/jboss/tools/cdk/ui/bot/test/CDKDevstudioBaseTest.java

```diff
@@ -0,0 +1,295 @@
+	
+	private static final String OPENSHIFT_USER_NAME = "openshift-dev";
+	
+	private static final String DOCKER_DAEMON_CONNECTION = SERVER_ADAPTER;
+	
+	private static final Logger log = Logger.getLogger(CDKDevstudioBaseTest.class);
+	
+	private static String USERNAME;
+	
+	private static String PASSWORD;
+	
+	// TODO: add validation of vagrant path
+	private static String VAGRANTFILE_PATH;
+	
+	static {
+		String username = System.getProperty("developers.username");
+		if (!(username == null || username.equals("") || username.startsWith("${"))){
+			USERNAME = username;
+		} else {
+			USERNAME = null;
+		}
+		String password = System.getProperty("developers.password");
+		if (!(password == null || password.equals("") || password.startsWith("${"))){
+			PASSWORD = password;
+		} else {
+			PASSWORD = null;
+		}
+		String vagrantfile = System.getProperty("vagrantfile.path");
+		if (!(vagrantfile == null || vagrantfile.equals("") || vagrantfile.startsWith("${"))){
+			VAGRANTFILE_PATH = vagrantfile;
+		} else {
+			VAGRANTFILE_PATH = null;
+		}	
+	}
+	
+	private static void checkCredentials() {
+		if (USERNAME == null || PASSWORD== null) {
+			throw new RedDeerException("Credentials for Red Hat Developers were not set properly");
+		}
+		log.info("Red Hat Developers username " + USERNAME + " and given password are set");
+		if (VAGRANTFILE_PATH == null) {
+			throw new RedDeerException("Path Vagrantfile path was not specified");
+		}
+		log.info("Vagrantfile path is set to " + VAGRANTFILE_PATH);
+	}
+	
+	@BeforeClass
+	public static void setUpEnvironemnt() {
+		log.info("Checking given program arguments");
+		checkCredentials();
+		log.info("Adding new Container Development Environment server adapter");
+		addNewCDEServer();
+	}
+	
+	@AfterClass
+	public static void tearDownEnvironment() {
+		checkCredentials();
+		log.info("Deleting Container Development Environment server adapter");
+		ServersView servers = new ServersView();
+		servers.open();
+		servers.getServer(SERVER_ADAPTER).delete(true);
+	}
+	
+	@Before
+	public void setUpServers() {
+		log.info("Open Servers view tab");
+		serversView = new CDEServersView();
+		serversView.open();
+		log.info("Getting server object from Servers View with name: " + SERVER_ADAPTER);
+		server = (CDEServer)serversView.getServer(SERVER_ADAPTER);
+		new WaitUntil(new JobIsRunning(), TimePeriod.SHORT, false);
+		log.info("Starting server adapter");
+		try {
+			server.start();
+		} catch (ServersViewException e) {
+			e.printStackTrace();
+		}
+		showCertificates();
+	}
+	
+	@After
+	public void tearDownServers() {
+		
+		if (server.getLabel().getState() == ServerState.STARTED) {
+			server.stop();
+		}
+		// remove SSL Certificate to be added at next server start at method annotated with before
+		deleteCertificates();
+		server = null;
+		serversView.close();
+	}
+	
+	@Test
+	public void testCDEIsAvailable() {
+		checkAvailableServers();
```

Comment:
I think you can save a considerable amount of time if the server would not start for this test case (this would mean, that it couldn't be done in @Before). Your call...